amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/KinesisClientLibConfigurator.java

```diff
@@ -62,14 +64,34 @@ public KinesisClientLibConfigurator() {
     public MultiLangDaemonConfiguration getConfiguration(Properties properties) {
         properties.entrySet().forEach(e -> {
             try {
-                utilsBean.setProperty(configuration, (String) e.getKey(), e.getValue());
+                Object value = e.getValue();
+                if(value instanceof String){
+                    value = ((String) value).trim();
+                }
+                utilsBean.setProperty(configuration, (String) e.getKey(), value);
             } catch (IllegalAccessException | InvocationTargetException ex) {
                 throw new RuntimeException(ex);
             }
         });
 
         Validate.notBlank(configuration.getApplicationName(), "Application name is required");
-        Validate.notBlank(configuration.getStreamName(), "Stream name is required");
+
+        try {
+            Validate.notBlank(configuration.getStreamArn(), "");
+            Arn streamArnObj = Arn.fromString(configuration.getStreamArn());
+
+            //Parse out the stream Name from the Arn (and/or override existing value for Stream Name)
+            String streamNameFromArn = streamArnObj.getResource().getResource();
+            configuration.setStreamName(streamNameFromArn);
+
+            //Parse out the region from the Arn and set (and/or override existing value for region)
+            String regionName = streamArnObj.getRegion();
+            Region regionObj = Region.of(regionName);
+            configuration.setRegionName(regionObj);
+        }catch (Exception e) {
```

Comment:
This will catch and swallow exceptions thrown from sdk when calling `Arn.fromString()` with an invalid Arn. IS this the error message we want to give to the user if, for example, they pass in an invalid stream Arn and no streamName?

Suggest only catching the `Validate.notBlank` exceptions and letting the SDK throw an error if an invalid Arn is passed.